diff --git a/init.c b/init.c
index d76d266..27dffce 100644
--- a/init.c
+++ b/init.c
@@ -4,7 +4,7 @@
 #include "stat.h"
 #include "user.h"
 #include "fcntl.h"
-//#define SINGLESHOT
+#define SINGLESHOT
 
 char *argv[] = { "sh", 0 };
 
@@ -21,33 +21,19 @@ main(void)
   dup(0);  // stderr
 
 #ifdef SINGLESHOT
-
-    // exec the tester and don't wait for it
+    // exec the tester and wait for it
     pid = fork();
     if(pid < 0){
       printf(1, "init: fork failed\n");
       exit();
     }
-    //if (pid == 0) {
-    //    exec("open_files", argv);
-    //    printf(1, "init: exec sh failed\n");
-    //    exit();
-    //}
-
-    sleep(100);
-    // exec the lsproc
-    pid = fork();
-    if (pid < 0) {
-        printf(1, "init: fork failed\n");
-        exit();
-    }
+    
     if (pid == 0) {
         exec("lsproc", argv);
         printf(1, "init: exec sh failed\n");
         exit();
     }
     wait();
-    wait();
     halt();
 #endif
 
